sdk/src/types/block/output/account.rs

```diff
@@ -388,6 +388,12 @@ impl AccountOutput {
         self.features.block_issuer().is_some()
     }
 
+    /// Returns whether the output can claim rewards based on its current and next state in a transaction.
+    pub fn can_claim_rewards(&self, next_state: Option<&Self>) -> bool {
+        // TODO: should destroying an account add rewards?
```

Comment:
Not super clear to me, I'll ask tomorrow

Reply:
Actually, TIP says the following so I guess so

- All of the following conditions hold:
        - The _Staking Feature_ is removed on the output side of the transaction or the account is destroyed.
        - A _Reward Input_ must be present whose `Index` references the account input.
        - The Mana amount on the input side of the transaction is increased by at most the amount defined in
          [TIP-40 (Validator Rewards)](../TIP-0040/tip-0040.md#validator-rewards).

Reply:
oh okay